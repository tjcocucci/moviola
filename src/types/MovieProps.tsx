export default interface MovieProps {
  id: number;
  title: string;
  poster_path: string;
}
