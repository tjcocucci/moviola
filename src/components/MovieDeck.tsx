import ImageCard from '@/components/ImageCard';
import MovieProps from '@/types/MovieProps';

const IMAGE_PATH = 'https://image.tmdb.org/t/p/w500';

export default function MovieDeck({ movies }: { movies: MovieProps[] }) {
  return (
    <div className="flex flex-wrap justify-center">
      {movies &&
        movies.map((movie: any) => (
          <ImageCard
            key={movie.id}
            imageUrl={
              movie.poster_path
                ? IMAGE_PATH + movie.poster_path
                : 'no-image.svg'
            }
            title={movie.title}
          />
        ))}
    </div>
  );
}
