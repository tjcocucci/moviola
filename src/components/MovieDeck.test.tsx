import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import MovieDeck from './MovieDeck';
import MovieProps from '@/types/MovieProps';

const defaultProps = {
  movies: [
    {
      id: 1,
      title: 'test',
      poster_path: 'test',
    },
    {
      id: 2,
      title: 'test2',
      poster_path: 'test2',
    },
  ],
};

const emptyProps = {
  movies: [],
};

describe('[ components / MovieDeck ]', () => {
  describe('when component is rendered', () => {
    it('should render snapshoted MovieDeck', () => {
      const { asFragment } = render(<MovieDeck {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when component is rendered with no movies', () => {
    it('should render snapshoted empty MovieDeck', () => {
      const { asFragment } = render(<MovieDeck {...emptyProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
