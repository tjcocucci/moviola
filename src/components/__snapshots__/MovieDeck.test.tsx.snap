// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ components / MovieDeck ] > when component is rendered > should render snapshoted MovieDeck 1`] = `
<DocumentFragment>
  <div
    class="container my-10 mx-auto  max-w-4xl"
  >
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4"
    >
      <div>
        <div
          class="relative drop-shadow shadow-white basis-1/2 lg:basis-1/4 xl:basis-1/5 border border-gray-200 rounded-lg overflow-hidden flex flex-col items-center justify-between h-full opacity-90 hover:opacity-100 hover:scale-105 "
        >
          <div
            class="relative flex-grow-0"
          >
            <div
              class="object-cover object-center drop-shadow-2xl shadow-white"
            >
              <img
                alt="test"
                data-nimg="1"
                decoding="async"
                height="450"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=640&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </div>
            <div>
              <h3
                class="w-full text-m font-bold p-4"
              >
                test
              </h3>
            </div>
          </div>
          <div
            class="w-full p-4 flex flex-col justify-stretch"
          >
            <button
              class="w-full text-white font-bold py-2 px-4 rounded hover:scale-105 hover:cursor-pointer bg-blue-500 hover:bg-blue-700"
            >
              Add to List
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="relative drop-shadow shadow-white basis-1/2 lg:basis-1/4 xl:basis-1/5 border border-gray-200 rounded-lg overflow-hidden flex flex-col items-center justify-between h-full opacity-90 hover:opacity-100 hover:scale-105 "
        >
          <div
            class="relative flex-grow-0"
          >
            <div
              class="object-cover object-center drop-shadow-2xl shadow-white"
            >
              <img
                alt="test2"
                data-nimg="1"
                decoding="async"
                height="450"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test2&w=640&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test2&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test2&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </div>
            <div>
              <h3
                class="w-full text-m font-bold p-4"
              >
                test2
              </h3>
            </div>
          </div>
          <div
            class="w-full p-4 flex flex-col justify-stretch"
          >
            <button
              class="w-full text-white font-bold py-2 px-4 rounded hover:scale-105 hover:cursor-pointer bg-blue-500 hover:bg-blue-700"
            >
              Add to List
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ components / MovieDeck ] > when component is rendered with no movies > should render snapshoted empty MovieDeck 1`] = `
<DocumentFragment>
  <div
    class="container my-10 mx-auto  max-w-4xl"
  >
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4"
    />
  </div>
</DocumentFragment>
`;
